src/main/java/mongodb/Main.java

```diff
@@ -0,0 +1,240 @@
+    // If they don't already exist, the driver and Atlas will create them
+    // automatically when you first write data.
+    String dbName = "myDatabase";
+    String collectionName = "recipes";
+
+    // a CodecRegistry tells the Driver how to move data between Java POJOs (Plain Old Java Objects) and MongoDB documents
+    CodecRegistry pojoCodecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+            fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+
+    // The MongoClient defines the connection to our MongoDB datastore instance (Atlas) using MongoClientSettings
+    // You can create a MongoClientSettings with a Builder to configure codecRegistries, connection strings, and more
+    MongoClientSettings settings = MongoClientSettings.builder()
+            .codecRegistry(pojoCodecRegistry)
+            .applyConnectionString(mongoUri).build();
+
+    MongoClient mongoClient;
+    try {
+       mongoClient = MongoClients.create(settings);
+    } catch (MongoException me) {
+      System.err.println("Unable to connect to the MongoDB instance due to an error: " + me);
+      return;
+    }
+
+    // MongoDatabase defines a connection to a specific MongoDB database
+    MongoDatabase database = mongoClient.getDatabase(dbName);
+    // MongoCollection defines a connection to a specific collection of documents in a specific database
+    MongoCollection<Recipe> collection = database.getCollection(collectionName, Recipe.class);
+
+    /*      *** INSERT DOCUMENTS ***
+     *
+     * You can insert individual documents using collection.insert().
+     * In this example, we're going to create 4 documents and then
+     * insert them all in one call with insertMany().
+     */
+
+    try {
+      // recipes is a static variable defined above
+      InsertManyResult result = collection.insertMany(recipes);
+      System.out.println("Inserted " + result.getInsertedIds().size() + " documents.");
+    } catch (MongoException me) {
+      System.err.println("Unable to insert any recipes into MongoDB due to an error: " + me);
+      return;
+    }
+
+    /*      *** FIND DOCUMENTS ***
+     *
+     * Now that we have data in Atlas, we can read it. To retrieve all of
+     * the data in a collection, we call find() with an empty filter. find()
+     * returns a cursor, which is a special kind of iterator that efficiently
+     * fetches documents in batches when they're requested. Here
+     * we use the try-with-resources pattern to automatically close the
+     * cursor once we finish reading the recipes.
+     */
+
+    try (MongoCursor<Recipe> cursor = collection.find().iterator()) {
+      while (cursor.hasNext()) {
+        Recipe currentRecipe = cursor.next();
+        System.out.println(String.format("%s has %d ingredients and takes %d minutes to make", currentRecipe.getName(), currentRecipe.getIngredients().size(), currentRecipe.getPrepTimeInMinutes()));
```

Comment:
You can use System.out.printf() instead of println(String.format()), but make sure to add the linefeeds manually.